src/lib/KeyStoreIndexeddb.js

```diff
@@ -14,20 +14,23 @@ class KeyStore {
         return this._instance;
     }
 
+    constructor() {
+        /** @type {Promise<IDBDatabase>|null} */
+        this._dbPromise = null;
+    }
+
     /**
      * @returns {Promise.<IDBDatabase>}
      * @private
      */
     async connect() {
-        if (this._db) return Promise.resolve(this._db);
+        if (this._dbPromise) return this._dbPromise;
 
-        return new Promise((resolve, reject) => {
+        this._dbPromise = new Promise((resolve, reject) => {
             const request = window.indexedDB.open(KeyStore.DB_NAME, KeyStore.DB_VERSION);
 
             request.onsuccess = () => {
-                /** @type {IDBDatabase | null} */
-                this._db = request.result;
-                resolve(this._db);
+                resolve(request.result);
             };
```

Comment:
Quite some overlap (aka c&p) with account store > extract common code in super class?

Reply:
I guess because the account store is deprecated. 
Are there going to be more stores? If no, ok.

Reply:
exactly, it's deprecated and thus I think it's good the way it is